### PR TITLE
DOC: Update of 'return_eigenvectors' description

### PR DESCRIPTION
**Pull request by akahard2dj**
Issued by @gauthe. #9082.

### CODE REVIEW

File: scipy/sparse/linalg/eigen/arpack/arpack.py [hunk 1]
```diff
@@ -1361,7 +1361,8 @@ def eigsh(A, k=6, M=None, sigma=None, which='LM', v0=None,
     Returns
     -------
     w : array
-        Array of k eigenvalues
+        Array of k eigenvalues.
+
```
Comment (ilayn) [inline scipy/sparse/linalg/eigen/arpack/arpack.py]: We don't need empty lines between the items in the same section.

File: scipy/sparse/linalg/eigen/arpack/arpack.py [hunk 2]
```diff
@@ -1446,7 +1447,13 @@ def eigsh(A, k=6, M=None, sigma=None, which='LM', v0=None,
     OPinv : N x N matrix, array, sparse matrix, or LinearOperator
         See notes in sigma, above.
     return_eigenvectors : bool
-        Return eigenvectors (True) in addition to eigenvalues
+        Return eigenvectors (True) in addition to eigenvalues. This value determines
+        the order in which eigenvalues are sorted.
+
+            If `return_eigenvectors` is True, eigenvalues are sorted by algebraic value
+
+            If `return_eigenvectors` is False, eigenvalues are sorted by magnitude value
```
Comment (ilayn) [inline scipy/sparse/linalg/eigen/arpack/arpack.py]: The term "absolute value" sounds more natural.

File: scipy/sparse/linalg/eigen/arpack/arpack.py (hunk 1 above)
Comment (akahard2dj) [reply scipy/sparse/linalg/eigen/arpack/arpack.py]: Yes, It seems that there is a blank line in editing process. I will correct your point.

File: scipy/sparse/linalg/eigen/arpack/arpack.py (hunk 2 above)
Comment (akahard2dj) [reply scipy/sparse/linalg/eigen/arpack/arpack.py]: I agree with you.
Generally speaking, 'absolution' is better that 'magnitude'.
Thanks for your quickly reviews, @ilayn.

File: scipy/sparse/linalg/eigen/arpack/arpack.py [hunk 3]
```diff
@@ -1442,11 +1442,26 @@ def eigsh(A, k=6, M=None, sigma=None, which='LM', v0=None,
...
+
+                If `return_eigenvectors` is False, eigenvalues are sorted by absolute value
+
+            For which = 'BE' or 'LA':
+                eigenvalues are always sorted by algebraic value
+
+            For which = 'SM':
+                If `return_eigenvectors` is True, eigenvalues are sorted by algebraic value
+
+                If `return_eigenvectors` is False, eigenvalues are sorted by decreasing absolute value.
```
Comment (jeffyancey) [inline scipy/sparse/linalg/eigen/arpack/arpack.py]: punctuation terminates this sentence, but not the ones on lines 1453, 1455, 1458, and 1461.

**Comment by jeffyancey** [comment]
FWIW, the docstring for the entire method is rampant with inconsistent punctuation. I don't consider that important, but now is not a bad time to make the docstring consistent.

File: scipy/sparse/linalg/eigen/arpack/arpack.py (hunk 3 above)
Comment (akahard2dj) [reply scipy/sparse/linalg/eigen/arpack/arpack.py]: I will revise the sentence with a punctuation termination.

**Comment by akahard2dj** [comment]
@jeffyancey I agree with you. I'll do more thorough checking and document work.

**Comment by akahard2dj** [comment]
@jeffyancey I fixed docstring consistency for a punctuation terminate.
Could you review this commit?

File: scipy/sparse/linalg/eigen/arpack/arpack.py [hunk 4]
```diff
@@ -1420,33 +1420,48 @@ def eigsh(A, k=6, M=None, sigma=None, which='LM', v0=None,
...
         The default value of 0 implies machine precision.
     Minv : N x N matrix, array, sparse matrix, or LinearOperator
-        See notes in M, above
+        See notes in M, above.
     OPinv : N x N matrix, array, sparse matrix, or LinearOperator
         See notes in sigma, above.
     return_eigenvectors : bool
-        Return eigenvectors (True) in addition to eigenvalues
+        Return eigenvectors (True) in addition to eigenvalues. This value determines
+        the order in which eigenvalues are sorted. The sort order is also dependent on the which variable.
```
Comment (jeffyancey) [inline scipy/sparse/linalg/eigen/arpack/arpack.py]: could you make which => `which`?

**Review by jeffyancey — COMMENTED** [review]
You'll have my approval as soon as which is marked as a variable with `'s

File: scipy/sparse/linalg/eigen/arpack/arpack.py (hunk 4 above)
Comment (akahard2dj) [reply scipy/sparse/linalg/eigen/arpack/arpack.py]: I fixed your request.
Changed general word to marked as a variable

**Review by jeffyancey — APPROVED** [review]

File: scipy/sparse/linalg/eigen/arpack/arpack.py [hunk 5]
```diff
@@ -1346,13 +1346,13 @@ def eigsh(A, k=6, M=None, sigma=None, which='LM', v0=None,
...
     generalized eigenvalue problem for w[i] eigenvalues
-    with corresponding eigenvectors x[i]
+    with corresponding eigenvectors x[i].
 
     Parameters
     ----------
     A : An N x N matrix, array, sparse matrix, or LinearOperator representing
-        the operation A * x, where A is a real symmetric matrix
-        For buckling mode (see below) A must additionally be positive-definite
+        the operation A * x, where A is a real symmetric matrix.
```
Comment (ev-br) [inline scipy/sparse/linalg/eigen/arpack/arpack.py]: Asterisks have special meaning in ReST, so should wrap A*x into double backticks.

Comment (ilayn) [reply scipy/sparse/linalg/eigen/arpack/arpack.py]: Also it's better to use ``A @ b`` for matrix multiplication.

**Comment by gauthe** [comment]
Line 1374 should be
`A @ x = w * M @ x.`
since w is a scalar in the definition of the generalized eigenvalue problem.

**Comment by akahard2dj** [comment]
@gauth I forgot that `w` was scala.
Thank for your valuable review.

**Comment by rgommers** [comment]
LGTM now, merging. Thanks @akahard2dj, all.

**Pull request merged**